src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -49,19 +52,9 @@ export const monoPlatform: Platform = {
     return new Promise<void>((resolve, reject) => {
       attachDebuggerHotkey(resourceLoader);
 
-      // dotnet.js assumes the existence of this
-      window['Browser'] = {
-        // eslint-disable-next-line @typescript-eslint/no-empty-function
-        init: () => { },
-      };
-
-      // Emscripten works by expecting the module config to be a global
-      // For compatibility with macOS Catalina, we have to assign a temporary value to window.Module
-      // before we start loading the WebAssembly files
-      addGlobalModuleScriptTagsToDocument(() => {
-        window['Module'] = createEmscriptenModuleInstance(resourceLoader, resolve, reject);
-        addScriptTagsToDocument(resourceLoader);
-      });
+      createEmscriptenModuleInstance(resourceLoader, resolve, reject);
+    }).then(() => {
+      window['Module'] = Module;
     });
```

Comment:
Instead of using `then` explicitly, could we use `await`? Just marginally cleaner.

For example, instead of returning `new Promise`, we could do something like:

```ts
return (async () => {
   ... code involving await ...
})();
```

Also I don't quite understand the original code because it's not returning the promise from `createEmscriptenModuleInstance`, so it will just call the `then` callback immediately won't it? Is that a typo?

Reply:
It's the `onReady();` executed from emscripten callback, see line 450. We could move the `new Promise` inside createEmscriptenModuleInstance, but it's not changing much about it.

Reply:
That clarifies things, thanks. I do think it would be much cleaner to make `createEmscriptenModuleInstance` return a regular promise (just by being an async function) instead of manually passing and calling `resolve`/`reject`.

Reply:
Improved